shared/editor/nodes/CodeFence.ts

```diff
@@ -134,12 +140,56 @@ export default class CodeFence extends Node {
           option.selected = node.attrs.language === value;
           select.appendChild(option);
         });
+        const codeClass = !node.attrs.diagram ? "editor-visible" : "";
+
+        const diagram = document.createElement("div");
+        if (node.attrs.language === "mermaidjs") {
+          const id = Math.round(Math.random() * 100000);
+          const diagramId = "mermaid-diagram-" + id;
+
+          diagram.classList.add("mermaid-diagram");
+          diagram.id = diagramId;
+          if (node.attrs.diagram) {
+            diagram.classList.add("diagram-visible");
+          }
+          diagram.setAttribute("contentEditable", "false");
+          try {
+            mermaid.mermaidAPI.render(diagramId, node.textContent, function (
```

Comment:
I'm worried at the performance impact of running this constantly, it's going to get triggered a lot – even when nothing changed in the diagram source. At the least we should probably get an idea of the performance hit here, but possibly move this to a plugin too so the rendering can react to changes in the node content.